### PR TITLE
solution

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_movies-list-fetch-movies/)

### CODE REVIEW

**Review by ZUBOV-ILLIA — CHANGES_REQUESTED** [review]
`4.Add a movie to the list after submitting a form and clear the input.`

You forget to cleare the input)

File: src/components/FindMovie/FindMovie.tsx [hunk 1]
```diff
@@ -1,54 +1,124 @@
...
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+          <div className="control">
+            <input
+              type="text"
+              id="movie-title"
+              placeholder="Enter a title to search"
+              className="input is-danger"
```
Comment (ZUBOV-ILLIA) [inline src/components/FindMovie/FindMovie.tsx:52]: `is-danger` must be only if wovie not finded )

**Review by ZUBOV-ILLIA — CHANGES_REQUESTED** [review]

**Comment by Denchepornyuk** [comment]
Screenshot 2022-06-20 at 14 21 37

show error if the input is empty and I click buttons

**Comment by Denchepornyuk** [comment]
Screenshot 2022-06-20 at 14 24 50

clear preview after adding film to the list

File: src/App.tsx [hunk 2]
```diff
@@ -12,6 +13,20 @@ export class App extends Component<{}, State> {
...
+        );
+      }
+
+      return (
+        { movies: [...prev.movies, theMovie] }
+      );
+    });
+  };
+
   render() {
```
Comment (Denchepornyuk) [inline src/App.tsx:30]: use functional components for this tasks

File: src/components/FindMovie/FindMovie.tsx [hunk 3]
```diff
@@ -1,54 +1,148 @@
...
-          </button>
+export const FindMovie: React.FC <Props> = ({ onAdd, ifMovieAdded }) => {
+  const urlMovies = 'https://www.omdbapi.com/?apikey=848ed1e8&';
+
+  const [theMovie, setTheMovie] = useState <Movie>({
+    Poster: '',
+    Title: '',
+    Plot: '',
+    imdbID: '',
+  });
```
Comment (Denchepornyuk) [inline src/components/FindMovie/FindMovie.tsx:20]:
```suggestion
  const [movie, setMovie] = useState <Movie | null>();
```
movie can be null

File: src/components/FindMovie/FindMovie.tsx [hunk 4]
```diff
@@ -1,54 +1,148 @@
...
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+export const FindMovie: React.FC <Props> = ({ onAdd, ifMovieAdded }) => {
+  const urlMovies = 'https://www.omdbapi.com/?apikey=848ed1e8&';
```
Comment (Denchepornyuk) [inline src/components/FindMovie/FindMovie.tsx:13]: move to api folder

File: src/components/FindMovie/FindMovie.tsx [hunk 5]
```diff
@@ -1,54 +1,148 @@
...
+  });
+  const [inputedMovie, setInputedMovie] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [added, setAdded] = useState(false);
+
+  const findMovie = () => {
+    fetch(`${urlMovies}t=${inputedMovie}`)
+      .then(response => {
+        response.json()
+          .then(movie => {
```
Comment (Denchepornyuk) [inline src/components/FindMovie/FindMovie.tsx:29]: move to api folder

**Review by Denchepornyuk — CHANGES_REQUESTED** [review]

File: src/components/FindMovie/FindMovie.tsx [hunk 6]
```diff
@@ -1,54 +1,158 @@
...
+    imdbID: '',
+  });
+  const [inputedMovie, setInputedMovie] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [added, setAdded] = useState(false);
+
+  const findMovie = () => {
+    fetch(`${urlMovies}t=${inputedMovie}`)
+      .then(response => {
+        response.json()
```
Comment (Denchepornyuk) [inline src/components/FindMovie/FindMovie.tsx:28]: move fetch to api folder

**Review by Denchepornyuk — CHANGES_REQUESTED** [review]
fix all previous comments

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**